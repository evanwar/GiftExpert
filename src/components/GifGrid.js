import  useFetchGifs  from "../hooks/useFetchGifs"
import GifGridItem from "./GifGridItem";
import PropTypes from 'prop-types';

const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>

            {loading && 'Cargando...'}

            <div className="card-grid">

                <ol>
                    {images.map(img => {
                        return <GifGridItem
                            key={img.id}
                            {...img} />;
                    })}
                </ol>
            </div>
        </>
    )
}


GifGrid.PropTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid;
