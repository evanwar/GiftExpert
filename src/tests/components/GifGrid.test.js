import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import  useFetchGifs  from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('Grid', () => {

    const category = 'Mario kart';

    test('Load template', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });


    test('debe de mostrar items cuando se cargan imagenes', () => {

        const gifs = [{ id: 'ABCD', url: 'https:www.google.com/cosa.jpg', title: 'Image' },{ id: 'ABC2', url: 'https:www.google.com/cosa.jpg', title: 'Image' }];


        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });


        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();

        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    });


})
