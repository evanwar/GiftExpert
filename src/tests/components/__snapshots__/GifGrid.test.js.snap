// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Load template 1`] = `
<Fragment>
  <h3>
    Mario kart
  </h3>
  Cargando...
  <div
    className="card-grid"
  >
    <ol />
  </div>
</Fragment>
`;

exports[`Grid debe de mostrar items cuando se cargan imagenes 1`] = `
<Fragment>
  <h3>
    Mario kart
  </h3>
  <div
    className="card-grid"
  >
    <ol>
      <GifGridItem
        id="ABCD"
        key="ABCD"
        title="Image"
        url="https:www.google.com/cosa.jpg"
      />
      <GifGridItem
        id="ABC2"
        key="ABC2"
        title="Image"
        url="https:www.google.com/cosa.jpg"
      />
    </ol>
  </div>
</Fragment>
`;
